### PR TITLE
feature/clients

### PR DESCRIPTION
**Pull request by dylanjongbloed**

### CODE REVIEW

File: src/components/clients/Clients.jsx [hunk 1]
```diff
@@ -7,6 +7,11 @@ import ClientForm from '../client-form/ClientForm';
 
 const Clients = ({ clients, createClient, fetchClients }) => {
   const [isClientFormVisible, setClientFormVisibility] = useState(false);
+  const [clientsSort, setClientsSort] = useState(clients);
+
+  const handleSort = sort => {
+    setClientsSort(clients.filter(client => client.name === sort.value));
```
Comment (RobbinHabermehl) [inline src/components/clients/Clients.jsx]: Having a `handleSort` that turns out to filter an array doesn't make sense.

File: src/components/clients/Clients.jsx [hunk 2]
```diff
@@ -35,8 +40,13 @@ const Clients = ({ clients, createClient, fetchClients }) => {
         >
           + New Client
         </button>
-        <select type="button" className={styles.sortClientsSelector}>
+        <select type="button" className={styles.sortClientsSelector} id="sort">
```
Comment (RobbinHabermehl) [inline src/components/clients/Clients.jsx]: What would you need the `id` for?

File: src/components/time-entries/TimeEntries.jsx [hunk 3]
```diff
@@ -42,6 +48,19 @@ function TimeEntries({
         isTimeEntryFormVisible={isTimeEntryFormVisible}
         toggleTimeEntryForm={toggleTimeEntryForm}
       />
+      <select
+        type="button"
```
Comment (RobbinHabermehl) [inline src/components/time-entries/TimeEntries.jsx]: Sort attributes alphabetically.

File: src/components/client-form/ClientForm.jsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+              value={address}
+            />
+          </label>
+        </div>
+        <div className={styles.formColumnTwo}>
+          <div className={styles.localityPostalCodeInputs}>
+            <label className={styles.labelPostalCode} htmlFor="postalCode">
+              Postal code
+              <input
+                className={`${styles.inputPostalCode} ${validity.postalCode ===
```
[the remark this answers is not in the document]
Comment (dylanjongbloed) [reply src/components/client-form/ClientForm.jsx]: When validity isn't set yet (before typing) or when validity checks have passed following user input, styles inputContactPhoneNumber will be used. This is desirable right?

File: src/redux/team-members/selectors.js [hunk 5]
```diff
@@ -22,6 +22,17 @@ export const teamMembersErrorSelector = createSelector(
 
 export const teamMembersByFieldSelector = createSelector(
   teamMembersRootSelector,
-  ({ items, sortByField }) =>
-    [...items].sort((a, b) => (a[sortByField] > b[sortByField] ? 1 : -1))
+  ({ items, sortByField, sortDirection }) =>
+    [...items].sort((a, b) => {
+      if (sortDirection === true) {
```
Comment (RobbinHabermehl) [inline src/redux/team-members/selectors.js]: Is the `=== true` really necessary?

It's not necessarily better, but you could also put that condition in the ternary:
`a[sortByField] > b[sortByField] ? (sortDirection ? 1 : -1) : (sortDirection ? -1 : 1);`
That way you can lose the `return` and return the expression immediately.

**Pull request merged**